### ISSUE TITLE
collections.namedtuple getting lost with DistributedDataParallel

### ISSUE THREAD

**Issue by carmocca** [pytorch/pytorch#44009]
## 🐛 Bug

When `DistributedDataParallel` is used, if the batch contains a `namedtuple`, it gets converted to a `Tuple`

Keeping the `namedtuple`s is necessary in my case because it is the object generated by my custom `collate_fn`

## To Reproduce

The culprit seems to be `torch.nn.parallel.scatter_gather.scatter`:

```python
NamedTuple = namedtuple("NamedTuple", ("a", "b"))

x = [(1, 2), NamedTuple(1, 2)]
y = torch.nn.parallel.scatter_gather.scatter(x, [0])
print(y)  # [[(1, 2), (1, 2)]]
```

## Expected behavior

`scatter` should reconstruct the `namedtuple`s

## Environment

PyTorch version: 1.6.0
Is debug build: False
CUDA used to build PyTorch: 10.2

OS: Pop!_OS 20.04 LTS (x86_64)
GCC version: (Ubuntu 9.3.0-10ubuntu2) 9.3.0
Clang version: Could not collect
CMake version: Could not collect

Python version: 3.8 (64-bit runtime)
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @xush6528 @osalpekar @jiayisuse @agolynski

**Comment by rohan-varma** [comment]
Related issues: https://github.com/pytorch/pytorch/issues/41327, https://github.com/pytorch/vision/issues/1048. Both have to do with lack of support for NamedTuples/dataclasses in scatter_gather APIs.

**Comment by tchaton** [comment]
Hey @rohan-varma,

Any updates on this one ?

Best,
T.C

**Comment by rohan-varma** [comment]
@tchaton Support for this has landed and should work on latest/nightly pytorch. Unfortunately it was landed a bit after 1.7.1, so you'd have to install from master/nightly - but it will be available as part of PT 1.8 general release.
